hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/execution/bulkinsert/TestBulkInsertInternalPartitioner.java

```diff
@@ -126,35 +134,40 @@ private void testBulkInsertInternalPartitioner(BulkInsertPartitioner partitioner
                                                  boolean isGloballySorted,
                                                  boolean isLocallySorted,
                                                  Map<String, Long> expectedPartitionNumRecords,
-                                                 Option<Comparator<HoodieRecord<? extends HoodieRecordPayload>>> comparator) {
+                                                 Option<Comparator<HoodieRecord<? extends HoodieRecordPayload>>> comparator,
+                                                 boolean populateMetaFields) {
     int numPartitions = 2;
-    JavaRDD<HoodieRecord<? extends HoodieRecordPayload>> actualRecords =
-        (JavaRDD<HoodieRecord<? extends HoodieRecordPayload>>) partitioner.repartitionRecords(records, numPartitions);
-    assertEquals(
-        enforceNumOutputPartitions ? numPartitions : records.getNumPartitions(),
-        actualRecords.getNumPartitions());
-    List<HoodieRecord<? extends HoodieRecordPayload>> collectedActualRecords = actualRecords.collect();
-    if (isGloballySorted) {
-      // Verify global order
-      verifyRecordAscendingOrder(collectedActualRecords, comparator);
-    } else if (isLocallySorted) {
-      // Verify local order
-      actualRecords.mapPartitions(partition -> {
-        List<HoodieRecord<? extends HoodieRecordPayload>> partitionRecords = new ArrayList<>();
-        partition.forEachRemaining(partitionRecords::add);
-        verifyRecordAscendingOrder(partitionRecords, comparator);
-        return Collections.emptyList().iterator();
-      }).collect();
-    }
+    if (!populateMetaFields) {
+      assertThrows(IllegalStateException.class, () -> partitioner.repartitionRecords(records, numPartitions));
```

Comment:
We can just return in here, there there would be no need to shift the block below